jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/RuntimeGroup.kt

```diff
@@ -17,108 +16,67 @@ import com.intellij.openapi.projectRoots.SdkType
 import com.intellij.openapi.util.KeyedExtensionCollector
-import com.intellij.openapi.util.LazyInstance
-import com.intellij.util.KeyedLazyInstance
-import com.intellij.util.xmlb.annotations.Attribute
 import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.core.IdBasedExtensionPoint
+
+/**
+ * IDs for built-in runtime groups that ship with toolkit
+ */
+object BuiltInRuntimeGroups {
+    const val Python = "PYTHON"
+    const val Dotnet = "DOTNET"
+    const val Java = "JAVA"
+    const val NodeJs = "NODEJS"
+}
 
 /**
  * Grouping of Lambda [Runtime] by parent language.
  *
- * A Lambda [Runtime] belongs to a single [RuntimeGroup], a [RuntimeGroup] may have several
- * Lambda [Runtime]s, [Language]s or [Sdk]s.
+ * A Lambda [Runtime] belongs to a single [RuntimeGroup], a [RuntimeGroup] may have several Lambda [Runtime]s, [Language]s or [Sdk]s.
  */
-enum class RuntimeGroup {
-    JAVA,
-    PYTHON,
-    NODEJS,
-    DOTNET;
-
-    private val info by lazy {
-        RuntimeGroupInformation.getInstances(this)
-    }
+interface RuntimeGroup {
+    val id: String
+    val runtimes: Set<Runtime>
+    val languageIds: Set<String>
 
-    val runtimes: Set<Runtime> by lazy { info.flatMap { it.runtimes }.toSet() }
-    val languageIds: Set<String> by lazy { info.flatMap { it.languageIds }.toSet() }
+    fun determineRuntime(project: Project): Runtime? = null
+    fun determineRuntime(module: Module): Runtime? = null
+    fun getModuleType(): ModuleType<*>? = null
+    fun getIdeSdkType(): SdkType? = null
+    fun supportsSamBuild(): Boolean = true
 
-    fun determineRuntime(project: Project): Runtime? = info.asSequence().mapNotNull { it.determineRuntime(project) }.firstOrNull()
-    fun determineRuntime(module: Module): Runtime? = info.asSequence().mapNotNull { it.determineRuntime(module) }.firstOrNull()
-    fun getModuleType(): ModuleType<*>? = info.asSequence().mapNotNull { it.getModuleType() }.firstOrNull()
-    fun getIdeSdkType(): SdkType? = info.asSequence().mapNotNull { it.getIdeSdkType() }.firstOrNull()
-    fun supportsSamBuild(): Boolean = info.asSequence().all { it.supportsSamBuild() }
+    companion object {
+        private val EP_NAM = ExtensionPointName.create<RuntimeGroup>("aws.toolkit.lambda.runtimeGroup")
 
-    internal companion object {
-        /**
-         * Lazily apply the predicate to each [RuntimeGroup] and return the first match (or null)
-         */
         @JvmStatic
-        fun find(predicate: (RuntimeGroup) -> Boolean): RuntimeGroup? = RuntimeGroup.values().asSequence().filter(predicate).firstOrNull()
+        fun find(predicate: (RuntimeGroup) -> Boolean): RuntimeGroup? = registeredRuntimeGroups().firstOrNull(predicate)
+
+        fun findById(id: String): RuntimeGroup? = find { it.id == id }
+
+        fun getById(id: String): RuntimeGroup = findById(id) ?: throw IllegalStateException("No RuntimeGroup with $id is registered")
```

Comment:
The distinction between find/get follows convention but it's not immediately obvious from the name. Can this be `get` and `getOrNull`?

Reply:
Ya